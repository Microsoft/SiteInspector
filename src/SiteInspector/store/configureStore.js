// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

/* eslint no-underscore-dangle: ["error", { "allow": ["__REDUX_DEVTOOLS_EXTENSION__"] }] */
import { applyMiddleware, createStore } from 'redux';
// eslint-disable-next-line import/no-extraneous-dependencies
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import throttle from 'lodash.throttle';
import { saveState, loadState } from './localStorage';
import createMainReducer from './reducers';

let middleware = applyMiddleware(thunk);

if (process.env.NODE_ENV !== 'production') {
  middleware = composeWithDevTools(middleware);
}

const store = createStore(createMainReducer(), loadState(), middleware);
store.asyncReducers = {};

export function injectAsyncReducer(storeToInject, asyncReducer) {
  storeToInject.asyncReducers = {
    ...storeToInject.asyncReducers,
    ...asyncReducer,
  };
  storeToInject.replaceReducer(createMainReducer(storeToInject.asyncReducers));
}

store.subscribe(throttle(() => {
  saveState(store.getState(), store.getState().ShellState.activeState);
}, 1000));

export default store;
