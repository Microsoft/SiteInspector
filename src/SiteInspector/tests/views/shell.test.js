import React from 'react';
import configureStore from 'redux-mock-store';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ShellContainer from '../../views/Shell/container';

describe('Shell view tests', () => {
  const initialState = {
    ShellState: {
      currentPath: '',
      position: 'right',
      tabs: [],
      visible: false,
      routesUpdated: false,
    },
    SharedState: {
      siteInspectorInitialized: true,
    },
  };
  const mockStore = configureStore();
  let store;
  let shell;

  configure({ adapter: new Adapter() });

  beforeEach(() => {
    store = mockStore(initialState);
    shell = shallow(<ShellContainer store={store} />);
  });

  it('Container was created', () => {
    expect(shell.length).toEqual(1);
  });

  it('SiteInspector rendered properly', () => {
    expect(shell).toMatchSnapshot();
  });
});
