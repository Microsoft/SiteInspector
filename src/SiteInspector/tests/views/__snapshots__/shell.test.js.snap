// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shell view tests SiteInspector rendered properly 1`] = `
<Shell
  currentPath=""
  messages={Array []}
  onToggleShell={[Function]}
  position="right"
  removeMessage={[Function]}
  resetRouteUpdate={[Function]}
  routesUpdated={false}
  setCurrentPath={[Function]}
  setShellPanelPosition={[Function]}
  siteInspectorInitialized={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  tabs={Array []}
  visible={false}
/>
`;
